File: scripts/check_db_status.py
#!/usr/bin/env python3
"""Check current database load status."""

import json
import os
from pathlib import Path
from urllib import error, request


def get_json(url: str, headers: dict) -> dict:
    """Fetch JSON from URL."""
    try:
        req = request.Request(url, headers=headers, method="GET")
        with request.urlopen(req, timeout=30) as response:
            return json.loads(response.read().decode("utf-8"))
    except error.URLError as e:
        print(f"❌ Connection error: {e}")
        return None


def main():
    # Load environment
    supabase_url = os.getenv("SUPABASE_URL")
    service_key = os.getenv("SUPABASE_SERVICE_ROLE_KEY")

    if not supabase_url:
        print("⚠️  SUPABASE_URL not set")
        print("Set it with: $env:SUPABASE_URL = 'https://<new-project-ref>.supabase.co'")
        return

    if not service_key:
        print("⚠️  SUPABASE_SERVICE_ROLE_KEY not set")
        print("Set it with: $env:SUPABASE_SERVICE_ROLE_KEY = 'your-key'")
        return
    
    headers = {
        "apikey": service_key,
        "Authorization": f"Bearer {service_key}",
        "Accept": "application/json"
    }
    
    print("=" * 60)
    print("📊 DATABASE STATUS CHECK")
    print("=" * 60)
    
    # Check knowledge_base
    print("\n📚 KNOWLEDGE_BASE:")
    kb_url = f"{supabase_url}/rest/v1/knowledge_base?select=count()&is_active=eq.true"
    kb_data = get_json(kb_url, headers)
    if kb_data:
        print(f"   Active docs: {kb_data[0]['count'] if kb_data else 'N/A'}")
    
    kb_total = f"{supabase_url}/rest/v1/knowledge_base?select=count()"
    kb_total_data = get_json(kb_total, headers)
    if kb_total_data:
        print(f"   Total docs: {kb_total_data[0]['count'] if kb_total_data else 'N/A'}")
    
    # Check legal_practice_kb
    print("\n🏛️  LEGAL_PRACTICE_KB:")
    practice_url = f"{supabase_url}/rest/v1/legal_practice_kb?select=count()&is_active=eq.true"
    practice_data = get_json(practice_url, headers)
    if practice_data:
        print(f"   Active docs: {practice_data[0]['count'] if practice_data else 'N/A'}")
    
    practice_total = f"{supabase_url}/rest/v1/legal_practice_kb?select=count()"
    practice_total_data = get_json(practice_total, headers)
    if practice_total_data:
        print(f"   Total docs: {practice_total_data[0]['count'] if practice_total_data else 'N/A'}")
    
    # Check legal_chunks
    print("\n📄 LEGAL_CHUNKS:")
    chunks_url = f"{supabase_url}/rest/v1/legal_chunks?select=count()"
    chunks_data = get_json(chunks_url, headers)
    if chunks_data:
        print(f"   Total chunks: {chunks_data[0]['count'] if chunks_data else 'N/A'}")
    
    # Check practice jobs
    print("\n⚙️  PRACTICE_CHUNK_JOBS:")
    jobs_pending = f"{supabase_url}/rest/v1/practice_chunk_jobs?select=count()&status=eq.pending"
    jobs_pending_data = get_json(jobs_pending, headers)
    if jobs_pending_data:
        print(f"   Pending jobs: {jobs_pending_data[0]['count'] if jobs_pending_data else '0'}")
    
    jobs_processing = f"{supabase_url}/rest/v1/practice_chunk_jobs?select=count()&status=eq.processing"
    jobs_proc_data = get_json(jobs_processing, headers)
    if jobs_proc_data:
        print(f"   Processing jobs: {jobs_proc_data[0]['count'] if jobs_proc_data else '0'}")
    
    jobs_completed = f"{supabase_url}/rest/v1/practice_chunk_jobs?select=count()&status=eq.completed"
    jobs_comp_data = get_json(jobs_completed, headers)
    if jobs_comp_data:
        print(f"   Completed jobs: {jobs_comp_data[0]['count'] if jobs_comp_data else '0'}")
    
    # Check legal_documents
    print("\n📖 LEGAL_DOCUMENTS:")
    docs_url = f"{supabase_url}/rest/v1/legal_documents?select=count()"
    docs_data = get_json(docs_url, headers)
    if docs_data:
        print(f"   Total documents: {docs_data[0]['count'] if docs_data else 'N/A'}")
    
    print("\n" + "=" * 60)


if __name__ == "__main__":
    main()

File: scripts/check-db-status.mjs
#!/usr/bin/env node
/**
 * Check database load status
 */

const SUPABASE_URL = process.env.SUPABASE_URL;
const SERVICE_KEY = process.env.SUPABASE_SERVICE_ROLE_KEY;

if (!SUPABASE_URL) {
  console.error('❌ SUPABASE_URL environment variable not set');
  process.exit(1);
}

if (!SERVICE_KEY) {
  console.error('❌ SUPABASE_SERVICE_ROLE_KEY environment variable not set');
  process.exit(1);
}

const headers = {
  'apikey': SERVICE_KEY,
  'Authorization': `Bearer ${SERVICE_KEY}`,
  'Accept': 'application/json',
};

async function fetchJson(url) {
  try {
    const response = await fetch(url, { headers, timeout: 30000 });
    if (!response.ok) {
      console.error(`HTTP ${response.status}: ${response.statusText}`);
      return null;
    }
    return await response.json();
  } catch (err) {
    console.error(`Fetch error: ${err.message}`);
    return null;
  }
}

async function main() {
  console.log('='.repeat(60));
  console.log('📊 DATABASE STATUS CHECK');
  console.log('='.repeat(60));

  // KNOWLEDGE_BASE
  console.log('\n📚 KNOWLEDGE_BASE:');
  const kbActive = await fetchJson(`${SUPABASE_URL}/rest/v1/knowledge_base?select=count()&is_active=eq.true`);
  if (kbActive?.length) {
    console.log(`   Active docs: ${kbActive[0].count}`);
  }

  const kbTotal = await fetchJson(`${SUPABASE_URL}/rest/v1/knowledge_base?select=count()`);
  if (kbTotal?.length) {
    console.log(`   Total docs: ${kbTotal[0].count}`);
  }

  // LEGAL_PRACTICE_KB
  console.log('\n🏛️  LEGAL_PRACTICE_KB:');
  const lpActive = await fetchJson(`${SUPABASE_URL}/rest/v1/legal_practice_kb?select=count()&is_active=eq.true`);
  if (lpActive?.length) {
    console.log(`   Active docs: ${lpActive[0].count}`);
  }

  const lpTotal = await fetchJson(`${SUPABASE_URL}/rest/v1/legal_practice_kb?select=count()`);
  if (lpTotal?.length) {
    console.log(`   Total docs: ${lpTotal[0].count}`);
  }

  // LEGAL_CHUNKS
  console.log('\n📄 LEGAL_CHUNKS:');
  const chunks = await fetchJson(`${SUPABASE_URL}/rest/v1/legal_chunks?select=count()`);
  if (chunks?.length) {
    console.log(`   Total chunks: ${chunks[0].count}`);
  }

  // LEGAL_DOCUMENTS
  console.log('\n📖 LEGAL_DOCUMENTS:');
  const docs = await fetchJson(`${SUPABASE_URL}/rest/v1/legal_documents?select=count()`);
  if (docs?.length) {
    console.log(`   Total documents: ${docs[0].count}`);
  }

  // PRACTICE_CHUNK_JOBS
  console.log('\n⚙️  PRACTICE_CHUNK_JOBS:');
  const pending = await fetchJson(`${SUPABASE_URL}/rest/v1/practice_chunk_jobs?select=count()&status=eq.pending`);
  if (pending?.length) {
    console.log(`   Pending jobs: ${pending[0].count}`);
  }

  const processing = await fetchJson(`${SUPABASE_URL}/rest/v1/practice_chunk_jobs?select=count()&status=eq.processing`);
  if (processing?.length) {
    console.log(`   Processing jobs: ${processing[0].count}`);
  }

  const completed = await fetchJson(`${SUPABASE_URL}/rest/v1/practice_chunk_jobs?select=count()&status=eq.completed`);
  if (completed?.length) {
    console.log(`   Completed jobs: ${completed[0].count}`);
  }

  console.log('\n' + '='.repeat(60));
}

main().catch(console.error);
